sql/catalyst/src/main/java/org/apache/spark/sql/util/CaseInsensitiveStringMap.java

```diff
@@ -78,11 +81,13 @@ public String get(Object key) {
 
   @Override
   public String put(String key, String value) {
+    originalMap.put(key, value);
```

Comment:
Things become a bit ambiguous here. E.g. after `put("a", "b")` and `put("A", "B")` to a empty `CaseInsensitiveStringMap`, `originalMap` has two new entries, while  `delegate` has only one new entry.
To me, this is the simplest way to maintain `originalMap`. Suggestions are welcomed.